### PR TITLE
N+1 fix and integration tests

### PR DESCRIPTION
**Pull request by Anand8317**
Solved the n+1 problem. Here is the before and after screenshots of logs
Screenshot 2024-01-31 at 9 53 01 AM

Screenshot 2024-01-31 at 10 55 27 AM

Integration tests for users and posts view files
Screenshot 2024-01-31 at 4 52 22 PM

Although this is a pair programming task, there will not be any commits from my coding partner.
Screenshot 2024-01-31 at 5 01 46 PM

### CODE REVIEW

File: spec/views/users/index.html.erb_spec.rb [hunk 1]
```diff
@@ -1,5 +1,31 @@
...
+  it 'displays usernames of all users' do
+    visit users_path
+    User.all.each do |user|
+      expect(page).to have_content(user.name)
+    end
+  end
+
+  it 'displays profile pictures of all users' do
+    visit users_path
+    User.all.each do |user|
```
Comment (thecodechaser) [inline spec/views/users/index.html.erb_spec.rb:13]: - [ ] The below tests are failing, Please make sure that all tests pass without any errors. (I can see you attached a screenshot in the PR description for test results, If tests are passing on your end Please record a small video showing passing tests and also show that all tests are present in the project while testing).

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Anand8317👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Good job fixing n+1 problem. ✔️
- Good job with integration tests. ✔️ 
- Giflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/users/index.html.erb_spec.rb (hunk 1 above)
Comment (Anand8317) [reply spec/views/users/index.html.erb_spec.rb:13]: Here is the video showing tests are passing on my end.
[Screen recording](https://share.icloud.com/photos/0e8VF8moEjQ37iRghsD5_Qhsg)

**Review by IngDavidTM — APPROVED** [review]
Hi Team 🙋‍♂️, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations!  🎉💯🌟 

## Top highlight 🎯

- The tests are passing ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 


## Optional suggestions

- [ ] Although your work is commendable 👍, please include instructions on running the tests either in the PR summary or the README file. I encountered difficulty getting your specs to function in my environment.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**